gr-blocks/lib/tcp_connection.cc

```diff
@@ -54,16 +54,25 @@ namespace gr {
     tcp_connection::send(pmt::pmt_t vector)
     {
       size_t len = pmt::blob_length(vector);
+
+      // Asio async_write() requires the buffer to remain valid until the handler is called. It will be
+      // deleted in tcp_connection::handle_write.
+      char* txbuf = new char[len];
```

Comment:
The handler is invoked if `io_service::run()` is called, which is part of this class' implicit contract since data is received via `async_read_some()`. As mentioned in the comment above, this could also be implemented with `boost::shared_ptr` or `std::unique_ptr` to guarantee destruction.